src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -269,6 +270,21 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
             $openapiOperation = $openapiOperation->withParameters($openapiParameters);
             $existingResponses = $openapiOperation->getResponses() ?: [];
             $overrideResponses = $operation->getExtraProperties()[self::OVERRIDE_OPENAPI_RESPONSES] ?? $this->openApiOptions->getOverrideResponses();
+            if ($operation instanceof HttpOperation && null !== $operation->getErrors()) {
+                foreach ($operation->getErrors() as $error) {
+                    /** @var ProblemExceptionInterface $exception */
+                    $exception = new $error();
```

Comment:
Not sure that we should call the constructor here, I'd suggest to use `$operation = $this->resourceMetadataCollectionFactory->create($error::class)->getOperation()` and to read `$operation->getStatus()` although I prefer the `getStatus` on that instance but I'm afraid we can't enforce that:
- the constructor has no arguments
- `getStatus` always exists (or we could've used Reflection->newInstanceWithoutConstructor)

Reply:
I might be mistaken, but won't `$this->resourceMetadataCollectionFactory->create($error::class)->getOperation()` throw a `OperationNotFoundException` for an ErrorResource ?
I do agree that the `new $error()` is itself error-prone if the constructor has arguments, I had not thought of this

Reply:
`ErrorResources` declares an ApiResource with an Error operation.